sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -255,156 +255,162 @@ export class MessageSender extends LinkEntity {
   private _trySend(encodedMessage: Buffer, sendBatch?: boolean): Promise<void> {
     const sendEventPromise = () =>
       new Promise<void>(async (resolve, reject) => {
-        let waitTimer: any;
-        log.sender(
-          "[%s] Sender '%s', credit: %d available: %d",
-          this._context.namespace.connectionId,
-          this.name,
-          this._sender!.credit,
-          this._sender!.session.outgoing.available()
-        );
-        if (!this._sender!.sendable()) {
-          log.sender(
-            "[%s] Sender '%s', waiting for 1 second for sender to become sendable",
-            this._context.namespace.connectionId,
-            this.name
-          );
-
-          await delay(1000);
+        try {
```

Comment:
There is an internal try/catch to cover the actual send operation `this._sender!.send`. Can we reduce the scope of this try to just `open()`?

Reply:
We could but I would suggest we leave it.

The issue is that it's difficult to inspect that you've messed this up (as I inadvertently proved by nobody seeing the error of not reject'ing the error explicitly, including myself).

In some of these changes I'm just saying "look, these are too hard to spot and being fine-grained here just makes it difficult to refactor".